engine/difficulty.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <string>
+
+namespace pong
+{
+
+class Difficulty
+{
+public:
+    enum class Level
+    {
+        Easy,
+        Normal,
+        Hard,
+        Insane
+    };
+
+    static Level GetLevel();
+    static void SetLevel(Level level);
+
+private:
+    Difficulty() = delete;
+    Difficulty(const Difficulty&) = delete;
+    Difficulty& operator=(const Difficulty&) = delete;
+    Difficulty(Difficulty&&) = delete;
+    Difficulty& operator=(Difficulty&&) = delete;
+    ~Difficulty() = delete;
```

Comment:
interesting that all of these are deleted. Can you explain why you did it this way? (not critiquing, just curious)

Reply:
Yeah, it comes from my hesitancy to create another singleton class lol. I just made the level variable static so that it doesn't need to do the whole GetInstance() thing. At that point there isn't really a need in keeping the constructor around